src/test/java/seedu/agendum/commons/util/StringUtilTest.java

```diff
@@ -47,6 +46,43 @@ public void getDetails_nullGiven_assertionError(){
         thrown.expect(AssertionError.class);
         StringUtil.getDetails(null);
     }
+    
+    @Test
+    public void isValidFilePath(){
+        // non-absolute file paths
+        assertFalse(StringUtil.isValidFilePath(null));
+        assertFalse(StringUtil.isValidFilePath(""));
+        assertFalse(StringUtil.isValidFilePath("a"));
+        assertFalse(StringUtil.isValidFilePath("data/xml"));
+        assertFalse(StringUtil.isValidFilePath("data/.xml"));
+        assertFalse(StringUtil.isValidFilePath("data/ .xml"));
+        assertFalse(StringUtil.isValidFilePath("data /valid.xml"));
+        assertFalse(StringUtil.isValidFilePath(" data/valid.xml"));
+        assertFalse(StringUtil.isValidFilePath("data.xml/data.xml"));
 
+        assertTrue(StringUtil.isValidFilePath("a/a.xml"));
+        assertTrue(StringUtil.isValidFilePath("Program Files/data.xml"));
+        assertTrue(StringUtil.isValidFilePath("folder/some-other-folder/data.dat"));
+        
+        // absolute file paths
+        assertFalse(StringUtil.isValidFilePath("CC:/valid.xml"));
+        assertFalse(StringUtil.isValidFilePath("asd:/valid.xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/"));
+        assertFalse(StringUtil.isValidFilePath("C:/Program Files"));
+        assertFalse(StringUtil.isValidFilePath("C:/a"));
+        assertFalse(StringUtil.isValidFilePath("C:/data/xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/data/.xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/data/ .xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/data /valid.xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/ data/valid.xml"));
+        assertFalse(StringUtil.isValidFilePath("C:/data.xml/data.xml"));
+        assertFalse(StringUtil.isValidFilePath("1:/data.xml"));
 
+        assertTrue(StringUtil.isValidFilePath("C:/a/a.xml"));
+        assertTrue(StringUtil.isValidFilePath("C:/Program Files/data.xml"));
+        assertTrue(StringUtil.isValidFilePath("Z:/folder/some-other-folder/data.dat"));
+        assertTrue(StringUtil.isValidFilePath("a:/folder/some-other-folder/data.dat"));
+        
```

Comment:
Do you mean something like /usr/data.xml
Not really sure about unix/max absolute paths